src/ripple/basics/tagged_integer.h

```diff
@@ -157,6 +163,19 @@ class tagged_integer
         return tagged_integer (lhs.m_value - rhs);
     }
 
+    template <class OtherInt>
+    friend
+    typename std::enable_if <
+        std::is_integral <OtherInt>::value &&
+            sizeof (OtherInt) <= sizeof (Int),
```

Comment:
@bachase: I'm not sure if it's worth it. We'd be almost venturing into the safe numerics style libraries like https://safeint.codeplex.com/ or https://github.com/robertramey/safe_numerics. It's surprisingly tricky and challenging code.